apps/dashr-pk-calc/utils/utils.R

```diff
@@ -0,0 +1,204 @@
+library(tidyr)
+library(data.table)
+library(pracma)
+
+GenerateParentList <- function(time_points) {
+# Generates flexible # of columns for the data-table
+  
+  parentList <- list(list())
+  parentList[[1]] <- list(
+    "name" = "Time (hr)",
+    "id" = "time",
+    "type" = "numeric")
+  
+  for (subject in seq(0, as.numeric(time_points) - 1)) {
+    
+    parentList[[length(parentList) + 1]] <- parentList[[length(parentList)]]
+    # Extend List length by copying last element
+    
+    parentList[[length(parentList)]]$name <- sprintf("Subj%s Conc (uM)", subject + 1)
+    parentList[[length(parentList)]]$id <- subject
+    parentList[[length(parentList)]]$type <- "numeric"
+    # Assign the values
+  }
+  return(parentList)
+}
+
+PkData2DashT <- function(df) {
+  # Converts raw df to nested list format 
+  
+  dfReshaped <- reshape(df[,-1], direction = "wide",
+                        idvar = "time", timevar = "subject_index")
+  # Reshape df to time subj0, sub1, subj2 ... subjn format
+  
+  names(dfReshaped) <- gsub("conc.", "", names(dfReshaped))
+  
+  return(df_to_list(dfReshaped))
+  # df_to_list is defined in dashTable
+}
+
+GenerateResultColumn <- function(results) {
+  # Generates flexible # of columns for the results-table
+
+  resultsRaw <- rbindlist(results)
+  # Convert results to df
+  resultsDf <- as.data.frame(sapply(resultsRaw, function(x) as.numeric(as.character(x))))
+  # Convert non-numerics to NA
+  resultsDf <- resultsDf[,colSums(is.na(resultsDf)) < nrow(resultsDf)]
+  # Remove Column if all NA
+  
+  resultList <- list(list())
+  resultList[[1]] <- list(
+    "name" = "Parameter",
+    "id" = "parameter",
+    "type" = "character")
+  
+  for (subject in as.numeric(names(resultsDf)[1:(ncol(resultsDf) -1)])) {
+    
+    resultList[[length(resultList) + 1]] <- resultList[[length(resultList)]]
+    # Extend List length by copying last element
+    
+    resultList[[length(resultList)]]$name <- sprintf("Subj%s", subject + 1)
+    resultList[[length(resultList)]]$id <- subject
+    resultList[[length(resultList)]]$type <- "numeric"
+    # Assign the values
+  }
+  resultList[[length(resultList) + 1]] <- resultList[[length(resultList)]]
+  resultList[[length(resultList)]]$name <- "Mean"
+  resultList[[length(resultList)]]$id <- "mean"
+  resultList[[length(resultList)]]$type <- "numeric"
+  # Append Mean column
+  
+  resultList[[length(resultList) + 1]] <- resultList[[length(resultList)]]
+  resultList[[length(resultList)]]$name <- "StDev"
+  resultList[[length(resultList)]]$id <- "stdev"
+  resultList[[length(resultList)]]$type <- "numeric"
+  # Append StDev column
+
+  return(resultList)
+}
+
+RowSD <- function(x) {
+# RowSD fun for mutating standard devitation
+# Helper FUN #1 for GenerateResultData
+  return(sqrt(rowSums((x - rowMeans(x))^2)/(dim(x)[2] - 1)))
```

Comment:
If I'm not mistaken, this can be done by applying `sd`:

```
return(apply(x, 1, sd))
```

Reply:
@sacul-git Thanks a lot for reviewing the app. I implemented all the changes with #acf2a20